2_process/src/munge_GFv1_catchments.R

```diff
@@ -0,0 +1,147 @@
+find_intersecting_hru <- function(prms_line,prms_hrus){
+  #' 
+  #' @description This function finds all of the HRU polygons that intersect a
+  #' PRMS segment, and returns the HRU ID (i.e., hru_segment) that has the
+  #' greatest percent of overlap with the PRMS segment
+  #' 
+  #' @param prms_line sf object containing one PRMS river segment of interest
+  #' prms_line must contain variables subsegid and geometry
+  #' @param prms_hrus sf (multi)polygon containing the HRU's from the NHGFv1
+  #
+  
+  # Transform PRMS segment and HRU's into a consistent projection
+  # using Albers Equal Area Conic CONUS
+  prms_line_proj <- sf::st_transform(prms_line, 5070)
+  prms_hrus_proj <- sf::st_transform(prms_hrus, 5070)
+  
+  # Calculate the length of prms_line and add as an attribute
+  prms_line_proj$length = sf::st_length(prms_line_proj$geometry)
+  
+  # Return the intersection between prms_line and HRU polygons 
+  # suppress warnings that "attribute variables are assumed to be spatially constant
+  # throughout all geometries"
+  intsct_hrus <- sf::st_intersection(prms_line_proj,prms_hrus_proj) %>%
```

Comment:
Reading just the variable name, it sounds like the intersected HRU polygons, but the intersection returns line segments. Could rename to `prms_line_intsct_hrus` but that's a long variable name

Reply:
Yeah, you're definitely right that the variable name implies that polygons will be returned rather than line segments. I think the more informative variable name is worth avoiding confusion for my future self 😃 . I've renamed the resulting intersected lines as `prms_line_intsct_hrus`.